### PR TITLE
Fix/upstream reviews3

### PR DESCRIPTION
**Pull request by plbossart**
This PR addresses most of the feedback from @andy-shev (Andy Shevchenko) on the core parts only.
Reviewers, I would appreciate if you can prioritize and spend a bit of time double-checking the fixes. Also there are 4 issues I added when I don't have the answers on my own.
I realize this is painful work, but I have no intention of doing it alone while you have fun with real work... I will merge these fixes on Thursday latest to get a chance for one round of validation before the next batch.
Thank you.
-Pierre

### CODE REVIEW

**Comment by ranj063** [comment]
@plbossart you're not alone :) I will prioritize this for tonight

File: sound/soc/sof/hw-spi.c [hunk 1]
```diff
@@ -158,7 +158,7 @@ static irqreturn_t spi_irq_handler(int irq __maybe_unused, void *context)
 	// on SPI based devices this will likely come via a SoC GPIO IRQ
 
 	// check if GPIO is assetred and if so run thread.
-	if (sof_pdata->gpio >= 0 &&
+	if (sof_pdata->gpio > 0 &&
```
Comment (RanderWang) [inline sound/soc/sof/hw-spi.c]: sof_pdata->gpio = desc_to_gpio(gpiod)

int desc_to_gpio(const struct gpio_desc *desc)
{
	return desc->gdev->base + (desc - &desc->gdev->descs[0]);
}

so gpio should be the type of in, not unsigned in

Comment (ranj063) [reply sound/soc/sof/hw-spi.c]: @RanderWang but is it ever likely to return anything < 0?

Comment (ranj063) [reply sound/soc/sof/hw-spi.c]: @RanderWang but is it ever likely to return anything < 0?

Comment (bardliao) [reply sound/soc/sof/hw-spi.c]: @ranj063 @RanderWang How about using gpio_is_valid(sof_pdata->gpio)? And, of cause sof_pdata->gpio should be int.

File: sound/soc/sof/sof-spi-dev.c [hunk 2]
```diff
@@ -22,12 +22,14 @@
 #include "hw-spi.h"
 #include "ops.h"
 
+#if 0
```
Comment (singalsu) [inline sound/soc/sof/sof-spi-dev.c]: Why "#if 0", can this part be just deleted?

Comment (keyonjie) [reply sound/soc/sof/sof-spi-dev.c]: I think we can delete this, and add it back when PM on it is supported.

File: sound/soc/sof/control.c [hunk 3]
```diff
@@ -392,8 +392,9 @@ int snd_sof_bytes_ext_get(struct snd_kcontrol *kcontrol,
 
 	/* Prevent read of other kernel data or possibly corrupt response */
 	data_size = cdata->data->size + sizeof(const struct sof_abi_hdr);
-	max_size = (size < max_size) ? size : max_size;
-	max_size = (be->max < max_size) ? be->max : max_size;
+	if (size < max_size) /* type warnings with min(), use test */
```
Comment (keyonjie) [inline sound/soc/sof/control.c]: here we are comparing with size(usigned int), so let's use type cast, or change to uint for max_size and be->max, to remove warning. 
And, we can use min3(a, b, c) to get that in one line.

File: sound/soc/sof/debug.c [hunk 4]
```diff
@@ -41,11 +41,11 @@ static ssize_t sof_dfsentry_read(struct file *file, char __user *buffer,
 		return -EINVAL;
 	if (pos >= size || !count)
 		return 0;
-	if (count > size - pos)
+	if (count > size - pos) /* type warnings with min(), use test */
```
Comment (keyonjie) [inline sound/soc/sof/debug.c]: count = min(count, (size_t) (size - pos)); ?

File: sound/soc/sof/trace.c [hunk 5]
```diff
@@ -75,7 +75,7 @@ static ssize_t sof_dfsentry_trace_read(struct file *file, char __user *buffer,
 	lpos_64 = lpos;
 	lpos = do_div(lpos_64, buffer_size);
 
-	if (count > buffer_size - lpos)
+	if (count > buffer_size - lpos) /* type warnings with min(), use test */
 		count = buffer_size - lpos;
```
Comment (keyonjie) [inline sound/soc/sof/trace.c]: ditto.

File: sound/soc/sof/hw-spi.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/hw-spi.c]: thanks for the comments, I just changed the type in sof.h, it's enough to get the warnings out of the way.

File: sound/soc/sof/sof-spi-dev.c (hunk 2 above)
Comment (plbossart) [reply sound/soc/sof/sof-spi-dev.c]: done

File: sound/soc/sof/control.c (hunk 3 above)
Comment (plbossart) [reply sound/soc/sof/control.c]: yes. done

File: sound/soc/sof/debug.c (hunk 4 above)
Comment (plbossart) [reply sound/soc/sof/debug.c]: done, thanks

File: sound/soc/sof/trace.c (hunk 5 above)
Comment (plbossart) [reply sound/soc/sof/trace.c]: done.

**Pull request merged**